internal/users/manager.go

```diff
@@ -119,6 +139,11 @@ func (m *Manager) UpdateUser(u users.UserInfo) (err error) {
 		return m.shouldClearDb(err)
 	}
 
+	// Update local groups.
+	if err := localgroups.UpdateLocalGroups(u.Name, localGroups); err != nil {
+		return errors.Join(err, m.shouldClearDb(m.cache.DeleteUser(u.UID)))
```

Comment:
we need a test with:
- multiple users in the database
- we fail to updatelocalgroups on ONE user
- we assert that we have the database with all the other users, but not that one.